File: dart/factory_method/operation/bin/operators.dart
abstract class Operators {
  double operate(double a, double b);
}

class Add implements Operators {
  @override
  double operate(double a, double b) {
    print(a + b);
    return a + b;
  }
}

class Sub implements Operators {
  @override
  double operate(double a, double b) {
    print(a - b);
    return a - b;
  }
}

class Mul implements Operators {
  @override
  double operate(double a, double b) {
    print(a * b);
    return a * b;
  }
}

class Div implements Operators {
  @override
  double operate(double a, double b) {
    print(a / b);
    return a / b;
  }
}

class Mod implements Operators {
  @override
  double operate(double a, double b) {
    print(a % b);
    return a % b;
  }
}

File: typescript/factory_method/operation/operators.ts
export interface Operators {
  operate(a: number, b: number): number
}

export class Add implements Operators {
  operate(a: number, b: number): number {
    console.log(a + b)
    return a + b
  }
}

export class Sub implements Operators {
  operate(a: number, b: number): number {
    console.log(a - b)
    return a - b
  }
}

export class Mul implements Operators {
  operate(a: number, b: number): number {
    console.log(a * b)
    return a * b
  }
}

export class Div implements Operators {
  operate(a: number, b: number): number {
    console.log(a / b)
    return a / b
  }
}

export class Mod implements  Operators {
  operate(a: number, b: number): number {
    console.log(a % b)
    return a % b
  }
}

